CONTRIBUTING.md

```diff
@@ -127,6 +127,245 @@ We try to follow the coding guidelines from the Go community.
 - Code should be commented
 - Code should pass all tests: `make test`
 
+# CLI Guidelines
+This is a design guideline used for development of Flow CLI. Purpose of this guidline
+is to achieve consistency across new features and allow composability of commands that
+build the fundaments of great CLI design.
+
+> Whatever software you’re building, you can be absolutely certain that people will
+> use it in ways you didn’t anticipate. Your software will become a part in a larger
+> system—your only choice is over whether it will be a well-behaved part. - [Clig](https://clig.dev/)
+
+
+# Human Interaction
+
+Our interaction with CLI is done by executing CLI commands and passing arguments
+and flags which need to be consistent. Consistency will allow interactions to
+be predictable and will with time feel natural even without resorting to reading instructions.
+
+## Commands
+Be consistent across commands and subcommands. In case we define a subcommand we
+should use `noun verb` pattern. **Don’t have ambiguous or similarly-named commands.**
+Naming and language should be same as used in Flow documentation. Use only
+lowercase letters, and dashes if you really need to. Keep it short.
+Users will be typing it all the time.
+
+```
+flow account
+```
+
+## Flags
+Flags are **named parameters**, denoted with either a hyphen, and a
+single-letter name (`-r`) or a double hyphen and a multiple-letter
+name (`--recursive`). Longer version is preferred for better readability
+and simpler learning experience in the beginning but after users will
+probably start using the shorter version. Every flag should have a shorter
+version and they should be both presented in the help. Every flag that can
+have a default value should have a default value so users don’t have to specify
+all of them when not needed. Support flags following a value delimited by a space or
+equal (=) sign (ex: `--username test` or `--username=test`)
+
+```
+flow account get --filter "address"
+```
+
+Use common naming for default falgs such as:
+
+`--log`: Logging level.
+
+`--output`: Output format (JSON, inline etc...).
+
+`-h, --help`: Help. This should only mean help. See the help section.
+
+`--version`: Version.
+
+
+## Arguments
+Arguments, or args, are positional parameters to a command. There should be as little
+arguments as possible because they make it hard for users to remember the sequence of arguments.
+Because they are relaying on position flags should be used where
+more than one argument would be required. Arguments should be one word
+verbs following general naming guideline. **Prefer flags to arguments**.
+
+Use an argument for the value that command requires in order to run.
```

Comment:
```suggestion
Arguments, or args, are positional parameters to a command. 
There should be as few arguments as possible, because it is hard for users to remember the sequence of arguments.

Because they are relying on position, flags should be used where more than one argument would be required. 

Arguments should be one word verbs following general naming guideline. 
**Prefer flags to arguments**.

Use an argument for the value that command requires in order to run.
```